src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateHistory = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        // Очищаем текущее состояние
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        // Добавляем свойства из extraData
+        currentState = { ...currentState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        // Удаляем указанные свойства
+        currentState = { ...currentState }; // Клонируем текущее состояние
```

Comment:
[CODE STYLE] - There's no need to make a copy of `currentState` using spread operator here since you're not making any changes to it before the loop in line 29. This line can be removed to improve code clarity and performance.